src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -4880,7 +4880,7 @@ else if (DataTypes.UNKNOWN_STREAM_LENGTH == dataLength)
                 isShortValue = columnPair.getValue().precision <= DataTypes.SHORT_VARTYPE_MAX_BYTES;
                 isNull = (null == currentObject);
                 if (currentObject instanceof String)
```

Comment:
should we not check isNull here ?

Reply:
no, if currentObject is an instance of String, it cannot be null, so we dont have to check isNull here.

Reply:
Turns out `null instanceOf <Class>` does not error out, so we can avoid null check